packages/@aws-cdk/assertions/lib/private/section.ts

```diff
@@ -1,54 +1,61 @@
 import { Match } from '../match';
 import { Matcher, MatchResult } from '../matcher';
 
-export type MatchSuccess = { match: true, matches: {[key: string]: any} };
-export type MatchFailure = { match: false, closestResult?: MatchResult, analyzedCount: number };
+export type MatchSuccess = { match: true, matches: { [key: string]: any }, analyzed: { [key: string]: any }, analyzedCount: number };
+export type MatchFailure = { match: false, closestResult?: MatchResult, analyzed: { [key: string]: any }, analyzedCount: number };
```

Comment:
I know we can technically remove `analyzedCount` here, but I kept it since it's being used in other places and convenient to use. 

If we want I can remove this field and update the other places to use `Object.keys(analyzed).length`